### PR TITLE
Add OpenAPI annotations for configuration api chemas

### PR DESCRIPTION
**Pull request by m4rk055**
Add descriptions and example values for `CreateSelectorPayload`, `CreateLegalBasePayload`, `CreateRetentionPolicy` and `CreateProvenancePayload` schemas.

### CODE REVIEW

File: modules/core/src/main/scala/api/messages/configuration/CreateLegalBasePayload.scala [hunk 1]
```diff
@@ -29,9 +30,19 @@ object ScopePayload {
 }
 
 case class CreateLegalBasePayload(
+    @description("type of the legal base")
+    @encodedExample("CONSENT")
     lbType: LegalBaseTerms,
```
Comment (noelmace) [inline modules/core/src/main/scala/api/messages/configuration/CreateLegalBasePayload.scala:35]: > « For Scala3 `enum`s, an implicit schema value has to be provided by hand.»
> https://tapir.softwaremill.com/en/latest/endpoint/schemas.html#enumerations

Just from a documentation perspective, I believe it is important to provide the list of accepted values as defined in PRIV. Right now, it only shows as a `string`.

> Also, I'm not even sure the validation is correctly handled here, as this docs points to https://tapir.softwaremill.com/en/latest/endpoint/validation.html... but that might be just me not knowing much about Tapir and Scala :sweat_smile: 

Adding an `externalDocs` entry could also be a good enough workaround. However, I have no idea of to add this in Tapir, as it seems to support it but didn't document this feature (see https://github.com/softwaremill/tapir/pull/75).

What would you propose to fix that here?

> The same comment applies to all other `enums` used in the API.

File: modules/core/src/main/scala/api/messages/configuration/CreateLegalBasePayload.scala [hunk 2]
```diff
@@ -29,9 +30,19 @@ object ScopePayload {
...
+    @description("legal base name")
+    @encodedExample("Contact form")
     name: Option[String],
+    @description("legal base description")
+    @encodedExample("Collection of the contact data for advertising")
     description: Option[String],
+    @description("legal base description")
```
Comment (noelmace) [inline modules/core/src/main/scala/api/messages/configuration/CreateLegalBasePayload.scala:42]: I believe this is a copy-paste typo? :slightly_smiling_face:

File: modules/core/src/main/scala/api/messages/configuration/CreateLegalBasePayload.scala (hunk 1 above)
Comment (noelmace) [reply modules/core/src/main/scala/api/messages/configuration/CreateLegalBasePayload.scala:35]: OK, my bad, I was totally wrong on this one :sweat_smile: :facepalm:

File: modules/core/src/main/scala/api/messages/configuration/CreateLegalBasePayload.scala (hunk 2 above)
Comment (m4rk055) [reply modules/core/src/main/scala/api/messages/configuration/CreateLegalBasePayload.scala:42]: Fixed

**Review by noelmace — APPROVED** [review]

**Pull request merged**